modules/widgets/blogs-i-follow/blogs-i-follow.php

```diff
@@ -0,0 +1,540 @@
+			// Process the JSON response, collecting the blavatars and the blog descriptions
+			foreach ($response_body as $site) {
+				if ( isset( $site->ID ) && isset( $site->icon ) && isset( $site->icon->img ) ) {
+					$blavatars[ $site->ID ] = '<img src="' . $site->icon->img . '" />';
+				}
+				if ( isset( $site->ID ) && isset( $site->description ) ) {
+					$descriptions[ $site->ID ] = $site->description;
+				}
+				if ( isset( $site->ID ) && isset( $site->name ) ) {
+					$blog_names[ $site->ID ] = $site->name;
+				}
+			}
+			// Update the subscriptions array with the additional fields retrieved from the API request
+			foreach ($subscriptions as &$subscription) {
+				$blog_id = $subscription['blog_id'];
+				$subscription['blavatar_img_tag'] = isset( $blavatars[ $blog_id ] ) ? $blavatars[ $blog_id ] : null;
+				$subscription['description'] = isset( $descriptions[ $blog_id ] ) ? $descriptions[ $blog_id ] : null;
+				$subscription['blog_name'] = isset( $blog_names[ $blog_id ] ) ? $blog_names[ $blog_id ] : null;
+			}
+			return $subscriptions;
+		}
+	}
+
+	function grid_view( $subscriptions ) {
+		wp_enqueue_style( 'hover-bubbles' );
+
+		// We are caching the HTML output because the blavatar functions
+		// make either queries or HTTP requests, so they are slow.
+		$output = get_transient( $this->id . '-widget' );
+
+		// TODO: Remove WP_DEBUG
+		if ( WP_DEBUG || empty( $output ) ) {
+
+			$output  = '';
+
+			$output .= "<div class='widgets-grid-layout no-grav'>";
+
+			$i = 0;
+			foreach ( $subscriptions as $subscription ) {
+				$i++;
+
+				if ( 'http://' === $subscription['blog_url'] )
+					$subscription['blog_url'] = $subscription['feed_url'];
+
+				$img = isset( $subscription['blavatar_img_tag'] ) ? $subscription['blavatar_img_tag'] : null;
+
+				if ( !$img ) {
+					if ( !empty( $subscription['blog_id'] ) ) {
+						// TODO: On WordPress.com, register this filter for get_blog_option. The admin email
+						// is not available via the REST API, so Jetpack currently cannot retrieve it.
+						$email = apply_filters( 'wpcom_blog_option', null, $subscription['blog_id'], 'admin_email' );
+						$http = is_ssl() ? 'https' : 'http';
+						$img = get_avatar( $email, self::$avatar_size, apply_filters( 'jetpack_static_url', esc_url_raw( $http . '://' . self::$default_avatar ) ) );
+					}
+				}
+				if ( !$img )
+					continue;
+
+				$blog_name = empty( $subscription['blog_name'] ) ? $this->get_inferred_blog_name( $subscription ) : $subscription['blog_name'];
+				$output .= "<div class='widget-grid-view-image wpcom-follow-gravatar'>";
+				$output .= "<a href='"  . esc_url( $subscription['blog_url'] ) . "' title='" . esc_attr( $blog_name ) . "' data-id='" . esc_attr( 'wpcom-bubble-' . $this->id . '-' . $i ) . "' class='bump-view' data-bump-view='bif'>";
+				$output .= $img;
+				$output .= "</a>";
+				$output .= "</div>";
+			}
+
+			$output .= "</div><div style='clear: both;'></div>";
```

Comment:
Yikes...looking into this there are other problems with the CSS (particular the responsive smaller layouts) I hadn't yet seen. Thanks for spurring me into checking that out.

As for this particular line, from what I can tell so far it isn't doing anything. Other CSS already clears the floats in this grid. I think I'm going to have to make larger updates, so if it still seems unnecessary I'll delete it as part of them (or integrate it if it is needed).